NuRadioReco/framework/base_trace.py

```diff
@@ -148,3 +155,82 @@ def deserialize(self, data_pkl):
+        rate.
+        """
+        # Some sanity checks
+        if not isinstance(x, BaseTrace):
+            raise TypeError('+ operator is only defined for 2 BaseTrace objects')
+        if self.get_trace() is None or x.get_trace() is None:
+            raise ValueError('One of the trace objects has no trace set')
+        if self.get_trace().ndim != x.get_trace().ndim:
+            raise ValueError('Traces have different dimensions')
+        trace_1 = copy.copy(self.get_trace())
+        trace_2 = copy.copy(x.get_trace())
+        if self.get_sampling_rate() != x.get_sampling_rate():
+            # Upsample trace with lower sampling rate
+            if self.get_sampling_rate() > x.get_sampling_rate():
+                sampling_rate = self.get_sampling_rate()
+                resampling_factor = fractions.Fraction(self.get_sampling_rate() / x.get_sampling_rate())
+                if (resampling_factor.numerator != 1):
+                    trace_2 = scipy.signal.resample(trace_2, resampling_factor.numerator * len(trace_2))
+                if(resampling_factor.denominator != 1):
+                    trace_2 = scipy.signal.resample(trace_2, len(trace_2) // resampling_factor.denominator)
+            else:
+                sampling_rate = x.get_sampling_rate()
+                resampling_factor = fractions.Fraction(x.get_sampling_rate() / self.get_sampling_rate())
+                if (resampling_factor.numerator != 1):
+                    trace_1 = scipy.signal.resample(trace_1, resampling_factor.numerator * len(trace_1))
+                if(resampling_factor.denominator != 1):
+                    trace_1 = scipy.signal.resample(trace_1, len(trace_1) // resampling_factor.denominator)
+        else:
+            sampling_rate = self.get_sampling_rate()
+
+        # Figure out which of the traces has the earlier trace start time
+        if self.get_trace_start_time() <= x.get_trace_start_time():
+            first_trace = trace_1
+            second_trace = trace_2
+            trace_start = self.get_trace_start_time()
+        else:
+            first_trace = trace_2
+            second_trace = trace_1
+            trace_start = x.get_trace_start_time()
+        # Calculate the difference in the trace start time between the traces and the number of
+        # samples that time difference corresponds to
+        time_offset = np.abs(x.get_trace_start_time() - self.get_trace_start_time())
+        i_start = int(round(time_offset * sampling_rate))
+        # We have to distinguish 2 cases: Trace is 1D (channel) or 2D(E-field)
+        # and treat them differently
+        if trace_1.ndim == 1:
+            # Calculate length the new trace needs to hold both input traces
+            trace_length = max(first_trace.shape[0], i_start + second_trace.shape[0])
+            # Make sure trace has an even number of samples
+            trace_length += trace_length % 2
```

Comment:
how does adding half of the trace length make the number of samples even? Don't you want to check if it is odd and then add `1` to the trace length?

Reply:
% is the modulo operator. It returns 0 if trace_length is even and 1 if it is odd. So that's basically already what I'm doing

Reply:
ups yes of course, I confused it with `//`